.github/workflows/npm-build.yml

```diff
@@ -0,0 +1,41 @@
+name: Lock npm packages
+on: 
+  push:
+    paths:
+      - 'yarn.lock'
+
+jobs:
+  build:
+    name: Build & Test
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js 14
+        uses: actions/setup-node@v2
+        with:
+          node-version: 14
+          registry-url: https://registry.npmjs.org/
+
+      - name: Remove old NPM lockfile
+        run: |
+          cd packages/hoprd
+          rm -f package-lock.json
+      - name: Install dependencies and create new NPM lockfile
+        run: |
+          cd packages/hoprd
+          npm install
```

Comment:
I would prefer if we use `npm i --package-lock-only` to generate the lock file

Reply:
Isn't the goal of this PR to generate an npm `lock` file? am not sure I understand why we shouldn't just use `npm i --package-lock-only`

Reply:
Yes, it *is* the goal. But half of it. We *can* generate a lockfile but it stays unclear if works or not because `npm` package resolution is slightly different from `yarn` package resolution

Reply:
is it? if package resolution is different then we need to investigate why - this would be a bug in yarn.

Reply:
Yes, it is, because of at least two reasons: hoisting and additional features such as custom package resolutions and workspaces. I wouldn't put too much effort in reverse-engineering yarn and leave this topic to yarn.